prog/weaver/metrics.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/weaveworks/weave/ipam"
+	"github.com/weaveworks/weave/nameserver"
+	"github.com/weaveworks/weave/net/address"
+	weave "github.com/weaveworks/weave/router"
+)
+
+func metricsHandler(router *weave.NetworkRouter, allocator *ipam.Allocator, ns *nameserver.Nameserver) http.Handler {
+	reg := prometheus.NewRegistry()
```

Comment:
What is the *harm* in exposing the stock metrics? Presumably there is enough structure to be able to clearly differentiate between generic metrics and weave net specific metrics, no?